### PR TITLE
Fix / Rageshake crashes + cleaning

### PR DESCRIPTION
**Pull request by BillCarsonFr**
https://github.com/matrix-org/riot-android-rageshakes/issues/5880
https://github.com/matrix-org/riot-android-rageshakes/issues/5877
https://github.com/matrix-org/riot-android-rageshakes/issues/5873
https://github.com/matrix-org/riot-android-rageshakes/issues/5871

### CODE REVIEW

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/MXOlmDevice.kt [hunk 1]
```diff
@@ -119,13 +114,13 @@ internal class MXOlmDevice @Inject constructor(
         try {
             deviceCurve25519Key = olmAccount!!.identityKeys()[OlmAccount.JSON_KEY_IDENTITY_KEY]
         } catch (e: Exception) {
-            Timber.e(e, "## MXOlmDevice : cannot find " + OlmAccount.JSON_KEY_IDENTITY_KEY + " with error")
+            Timber.e(e, """## MXOlmDevice : cannot find ${OlmAccount.JSON_KEY_IDENTITY_KEY} with error""")
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/MXOlmDevice.kt]: Triple " ?

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/algorithms/megolm/MXMegolmDecryption.kt [hunk 2]
```diff
@@ -64,19 +65,19 @@ internal class MXMegolmDecryption(private val credentials: Credentials,
...
-    private fun decryptEvent(event: Event, timeline: String, requestKeysOnFail: Boolean): MXEventDecryptionResult? {
+    private fun decryptEvent(event: Event, timeline: String, requestKeysOnFail: Boolean): MXEventDecryptionResult {
         val encryptedEventContent = event.content.toModel<EncryptedEventContent>()!!
         if (TextUtils.isEmpty(encryptedEventContent.senderKey) || TextUtils.isEmpty(encryptedEventContent.sessionId) || TextUtils.isEmpty(encryptedEventContent.ciphertext)) {
             throw MXDecryptionException(MXCryptoError(MXCryptoError.MISSING_FIELDS_ERROR_CODE,
                     MXCryptoError.UNABLE_TO_DECRYPT, MXCryptoError.MISSING_FIELDS_REASON))
         }
 
-        var eventDecryptionResult: MXEventDecryptionResult? = null
+        val eventDecryptionResult: MXEventDecryptionResult?
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/algorithms/megolm/MXMegolmDecryption.kt]: maybe move this declaration inside the 'if' block

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/algorithms/megolm/MXMegolmDecryption.kt [hunk 3]
```diff
@@ -197,22 +201,25 @@ internal class MXMegolmDecryption(private val credentials: Credentials,
...
-                ArrayList(forwardedRoomKeyContent.forwardingCurve25519KeyChain)
+
+            forwardedRoomKeyContent.forwardingCurve25519KeyChain?.let {
+                forwardingCurve25519KeyChain.addAll(it)
             }
+//            forwardingCurve25519KeyChain = if (forwardedRoomKeyContent.forwardingCurve25519KeyChain == null) {
+//                ArrayList()
+//            } else {
+//                ArrayList(forwardedRoomKeyContent.forwardingCurve25519KeyChain)
+//            }
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/algorithms/megolm/MXMegolmDecryption.kt]: Remove this commented out code

File: vector/src/main/java/im/vector/riotx/core/platform/VectorViewModel.kt [hunk 4]
```diff
@@ -21,4 +21,4 @@ import com.airbnb.mvrx.MvRxState
 import im.vector.riotx.BuildConfig
 
 abstract class VectorViewModel<S : MvRxState>(initialState: S)
-    : BaseMvRxViewModel<S>(initialState, debugMode = BuildConfig.DEBUG)
\ No newline at end of file
+    : BaseMvRxViewModel<S>(initialState, false)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/core/platform/VectorViewModel.kt]: error?

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/store/db/model/IncomingRoomKeyRequestEntity.kt [hunk 5]
```diff
@@ -33,9 +33,9 @@ internal open class IncomingRoomKeyRequestEntity(
 
     fun toIncomingRoomKeyRequest(): IncomingRoomKeyRequest {
         return IncomingRoomKeyRequest().apply {
-            requestId = requestId
-            userId = userId
-            deviceId = deviceId
+            requestId = this@IncomingRoomKeyRequestEntity.requestId
+            userId = this@IncomingRoomKeyRequestEntity.userId
+            deviceId = this@IncomingRoomKeyRequestEntity.deviceId
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/store/db/model/IncomingRoomKeyRequestEntity.kt]: Keep my version please, sorry for the conflict

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/MXOlmDevice.kt (hunk 1 above)
Comment (BillCarsonFr) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/MXOlmDevice.kt]: fixed

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/algorithms/megolm/MXMegolmDecryption.kt (hunk 2 above)
Comment (BillCarsonFr) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/algorithms/megolm/MXMegolmDecryption.kt]: yes

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/algorithms/megolm/MXMegolmDecryption.kt (hunk 3 above)
Comment (BillCarsonFr) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/algorithms/megolm/MXMegolmDecryption.kt]: done

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/store/db/model/IncomingRoomKeyRequestEntity.kt (hunk 5 above)
Comment (BillCarsonFr) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/store/db/model/IncomingRoomKeyRequestEntity.kt]: ok i reverted to error code, will be fixed with other merge

**Pull request merged**